### PR TITLE
Introduce IDynamicCommandBuilder to improve how commands are configured

### PR DESCRIPTION
**Pull request by jeanplevesque**
GitHub Issue: #

## Proposed Changes

- Feature
- Refactoring (no functional changes, no api changes) 

## What is the current behavior?

- It's hard to configure dynamic commands.
- We cannot easily "remove" a default strategy decorator locally.

## What is the new behavior?

- `IDynamicCommandBuilder` is used to easily manipulate strategies before the command creation.
- Decorators can be added and removed on a builder.
- It is now possible to locally add (in ViewModels) decorators that wraps all existing ones (default from the factory).

## Checklist

Please check if your PR fulfills the following requirements:

- [x] Documentation has been added/updated
- [x] Automated Unit / Integration tests for the changes have been added/updated
- [ ] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

**Review by vincentcastagna — APPROVED** [review]

File: src/DynamicMvvm.Abstractions/Command/IDynamicCommand.md [hunk 1]
```diff
@@ -147,59 +147,62 @@ public static class MyCustomCommandDecoratorExtensions
...
-          - CanExecuteStrategy (local decorator)
-            - ... (local decorator)
-              - ActionCommandStrategy (implementation)
+  - BackgroundCommandStrategy (default config)
+    - ErrorHandlerCommandStrategy (default config)
+      - LogCommandStrategy (default config)
+        - CanExecuteStrategy (local config)
+          - ... (other local strategies)
+            - ActionCommandStrategy (implementation)
 ```
```
Comment (MatFillion) [inline src/DynamicMvvm.Abstractions/Command/IDynamicCommand.md:207]: Should we have an example on how to "remove" a decorator?

File: README.md [hunk 2]
```diff
@@ -34,21 +34,35 @@ public class MyViewModel : ViewModelBase
...
-  // This will create a DynamicProperty with a name of "MyPropertyFromTask" and an initial value of 10.
+  // This will create and attach a new IDynamicProperty named "MyStringProperty"
+  // to your IViewModel. It's initial value will be "hello". It can be bound two-way.
+  public string MyStringProperty
+  {
+    get => this.Get(initialValue: "hello");
+    set => this.Set(value);
+  }
+
+  // This will create a DynamicProperty with a name of "MyAsyncProperty" and an initial value of 10.
```
Comment (YGuerin) [inline README.md:45]: "and an initial value of 20."

Comment (jeanplevesque) [reply README.md:45]: Actually the initial value is `default(int)` in this case. I'll update the doc!

File: src/DynamicMvvm.Abstractions/Command/IDynamicCommand.md (hunk 1 above)
Comment (jeanplevesque) [reply src/DynamicMvvm.Abstractions/Command/IDynamicCommand.md:207]: https://github.com/nventive/Chinook.DynamicMvvm/pull/13

File: README.md (hunk 2 above)
Comment (jeanplevesque) [reply README.md:45]: https://github.com/nventive/Chinook.DynamicMvvm/pull/13

**Pull request merged**